### PR TITLE
Remove redux from angular

### PR DESCRIPTION
**Pull request by eneufeld**
* Introduce `JSONFormsAngularService` which replaces redux
* Adapt renderers
* Update tests

### CODE REVIEW

**Comment by ruimarques** [comment]
Oh yes this would be awesome!

Anyway I can help?

To provide some context I am considering to add jsonforms to an angular application product but I will need:

- uncoupling from redux as we don't use it and I don't want to carry that dependency (we use other state mgmt lib)
- custom renderer which we will develop for our needs, we have our own components we want to use. (jsonforms already supports this I believe)

**Comment by sdirix** [comment]
@ruimarques Thanks for your interest! Rendering customization is a staple of JSON Forms on which the whole architecture is based. Actually all our provided renderers use the very same mechanisms we expect users of JSON Forms to use too.

If you'd like to help you could test this PR in your setup and see if we missed something when removing redux or you'd expect a different replacement interface (or implementation). The more eyes look at this, the better.

File: packages/angular-material/test/autocomplete-control.spec.ts [hunk 1]
```diff
@@ -99,104 +101,95 @@ describe('Autocomplete control Base Tests', () => {
   });
 
   it('should render', fakeAsync(() => {
-    const mockSubStore = setupMockStore(fixture, { uischema, schema, data });
-    mockSubStore.complete();
-    fixture.detectChanges();
+    setupMockStore(fixture, { uischema, schema, data });
+    getJsonFormsService(component).updateCore(
```
Comment (sdirix) [inline packages/angular-material/test/autocomplete-control.spec.ts]: Why do we need to do this? Doesn't `setupMockStore` not already setup the `JsonFormsService`?

File: packages/core/src/actions/index.ts [hunk 2]
```diff
@@ -56,6 +56,14 @@ export const SET_LOCALIZED_UISCHEMAS: 'jsonforms/SET_LOCALIZED_UISCHEMAS' =
 export const ADD_DEFAULT_DATA: 'jsonforms/ADD_DEFAULT_DATA' = `jsonforms/ADD_DEFAULT_DATA`;
 export const REMOVE_DEFAULT_DATA: 'jsonforms/REMOVE_DEFAULT_DATA' = `jsonforms/REMOVE_DEFAULT_DATA`;
 
+export type ValidCoreActions =
```
Comment (sdirix) [inline packages/core/src/actions/index.ts]: Shouldn't we just name them `CoreActions`?
```suggestion
export type CoreActions =
```

File: packages/core/src/actions/index.ts [hunk 3]
```diff
@@ -200,6 +208,10 @@ export const setConfig = (config: any) => (dispatch: Dispatch<AnyAction>) => {
   });
 };
 
+export type ValidUISchemaReducerActions =
```
Comment (sdirix) [inline packages/core/src/actions/index.ts]:
```suggestion
export type UISchemaActions =
```

File: packages/core/src/actions/index.ts [hunk 4]
```diff
@@ -231,6 +243,11 @@ export const unregisterUISchema = (
   };
 };
 
+export type ValidLocaleActions =
```
Comment (sdirix) [inline packages/core/src/actions/index.ts]:
```suggestion
export type LocaleActions =
```

File: packages/angular-material/src/other/master-detail/master.ts [hunk 5]
```diff
@@ -241,10 +242,9 @@ export class MasterListComponent extends JsonFormsArrayControl {
     this.removeItems(this.propsPath, [item])();
   }
 
-  protected mapToProps(state: JsonFormsState): ArrayControlProps {
+  protected mapToProps(state: JsonFormsState): StatePropsOfArrayControl {
```
[the remark this answers is not in the document]
Comment (sdirix) [reply packages/angular-material/src/other/master-detail/master.ts]: If we change this here then we should change all occurrences of this pattern

[the remark this answers is not in the document]
Comment (eneufeld) [reply packages/angular-material/src/other/master-detail/master.ts]: we can't as this is a method that we need to override

**Pull request merged**